io.openems.edge.goodwe.et/src/io/openems/edge/goodwe/et/ess/GoodWeEtBatteryInverterImpl.java

```diff
@@ -507,19 +505,49 @@ private void updatechannels() {
 		/*
 		 * Update Allowed charge and Allowed discharge
 		 */
+
+//		System.out.println(
+//				"Allowed Charge power: " + allowedChargePower + " Allowed Discharge Power: " + allowedDischargePower);
+
 		Integer soc = this.getSoc().get();
-		Integer maxApparentPower = this.getMaxApparentPower().get();
+		Integer maxBatteryPower = this.config.maxBatteryPower();
 
-		if (soc == null || soc >= 99) {
-			this._setAllowedChargePower(0);
-		} else {
-			this._setAllowedChargePower(TypeUtils.multiply(maxApparentPower, -1));
+		Integer allowedCharge = null;
+		Integer allowedDischarge = null;
+
+		if (productionPower == null) {
+			productionPower = 0;
```

Comment:
@venu-sagar: why do you initialize `productionPower` with `null` in the first place if you anyway set it to zero later?